### DISCUSSION TITLE
ISR gives permissions error on AWS (tried both serverless and Amplify)

### DISCUSSION THREAD

**Discussion by mindcross** [vercel/next.js#31405] [Help]
I created a very basic NextJS example app to demonstrate Incremental Static Regeneration and deployed it to AWS but when I attempt to access it via the browser I get a 503 error and this message:

"The Lambda function associated with the CloudFront distribution is invalid or doesn't have the required permissions...."

Then looking in the Cloudwatch logs for the Lamdas I found this:

"Access to the resource https://sqs.us-east-1.amazonaws.com/ is denied."

Accessing the home page works fine. But this error happens when accessing the dynamic route like [myapp.com]/product/1.

You can see the code for this page here:

[https://github.com/mindcross/aaron-next-test/blob/main/pages/product/%5Bproductid%5D.js](https://github.com/mindcross/aaron-next-test/blob/main/pages/product/%5Bproductid%5D.js)

The revalidate value is 10 and fallback is set to 'blocking'.

I have tried this using Serverless and also using AWS Amplify and they both produce the same permissions error message. From everything I've read so far, it seems that ISR should be supported and this is the absolutely simplest example I could create so I am confused why it doesn't work. 

I'm using NextJS version 11.1.2.

Has anyone gotten ISR to work deploying to AWS using Serverless or Amplify? Any help is appreciated!

**Comment by mingyeungs** [comment]
Came across the same issue today and found the solution here: https://github.com/aws-amplify/amplify-hosting/issues/2347

The issue takes place when you add ISR to a previously static app, and the simplest solution is to create a new Amplify app and connect again. Alternatively, disconnect the branch and reconnect again also work (need to wait for Amplify to clear resource though)
